go-controller/pkg/ovn/egressip.go

```diff
@@ -1000,6 +1028,19 @@ func (oc *Controller) addPodEgressIPAssignments(name string, statusAssignments [
 	return nil
 }
 
+// deleteAllocatorEgressIPAssignmentIfExists deletes egressIP config from node allocations map
+// if the entry is available and returns true, otherwise returns false.
+func (oc *Controller) deleteAllocatorEgressIPAssignmentIfExists(name, egressIP string) bool {
+	oc.eIPC.allocator.Lock()
+	defer oc.eIPC.allocator.Unlock()
+	_, existingAllocations := oc.getSortedEgressData()
+	if status, exists := existingAllocations[egressIP]; exists && status.Name == name {
+		delete(existingAllocations, egressIP)
```

Comment:
Does this here really have any effect?

getSortedEgressData builds a new slice every time it is called:
~~~
1497 // getSortedEgressData returns a sorted slice of all egressNodes based on the
1498 // amount of allocations found in the cache
1499 func (oc *Controller) getSortedEgressData() ([]*egressNode, map[string]egressIPNodeStatus) {
1500         assignableNodes := []*egressNode{}
1501         allAllocations := make(map[string]egressIPNodeStatus)
1502         for _, eNode := range oc.eIPC.allocator.cache {
1503                 if eNode.isEgressAssignable && eNode.isReady && eNode.isReachable {
1504                         assignableNodes = append(assignableNodes, eNode)
1505                 }
1506                 for ip, eipName := range eNode.allocations {
1507                         allAllocations[ip] = egressIPNodeStatus{Node: eNode.name, Name: eipName}
1508                 }
1509         }
1510         sort.Slice(assignableNodes, func(i, j int) bool {
1511                 return len(assignableNodes[i].allocations) < len(assignableNodes[j].allocations)
1512         })
1513         return assignableNodes, allAllocations
1514 }
1515 
1516 func (oc *Controller) setNodeEgressAssignable(nodeName string, isAssignable bool) {
1517         oc.eIPC.allocator.Lock()
1518         defer oc.eIPC.allocator.Unlock()
1519         if eNode, exists := oc.eIPC.allocator.cache[nodeName]; exists {
 NORMAL  ᚠ bc68cc4  ~/development/ovn-kubernetes/go-controller/pkg/ovn/egressip.go
~~~

If we delete egressIP from that newly built slice, but then never use the newly built slice ... there's no effect? Or am I reading that wrongly?

Reply:
If the delete is an operation without any effect (what I think it is), then  the method should be renamed from `deleteAllocatorEgressIPAssignmentIfExists` to `doesAllocatorEgressIPAssignmentExist` or something like that

Reply:
Also look at the original method, which deletes directly from the allocator cache - that's IMO different than what you are doing because it operatoes on the cache, and not on a new slice that is built from the cache
~~~
1003 // deleteAllocatorEgressIPAssignments deletes the allocation as to keep the
1004 // cache state correct, also see addAllocatorEgressIPAssignments
1005 func (oc *Controller) deleteAllocatorEgressIPAssignments(statusAssignments []egressipv1.EgressIPStatusItem) {
1006         oc.eIPC.allocator.Lock()
1007         defer oc.eIPC.allocator.Unlock()
1008         for _, status := range statusAssignments {
1009                 if eNode, exists := oc.eIPC.allocator.cache[status.Node]; exists {
1010                         delete(eNode.allocations, status.EgressIP)
1011                 }
1012         }
1013 }
~~~

Reply:
@andreaskaris good catch. thanks. we intend to delete unassigned egress ip entry from node allocator cache, but unfortunately i didn't have a closer look on getSortedEgressData method. yes, it returns new slice. 
so now directly deleting it from allocator cache.